packages/core/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -905,6 +907,17 @@ public void fetchNativeDeviceContexts(Promise promise) {
     }
 
     final @Nullable IScope currentScope = InternalSentrySdk.getCurrentScope();
+    if (currentScope != null) {
+      // Remove react-native breadcrumbs
+      Iterator<Breadcrumb> breadcrumbsIterator = currentScope.getBreadcrumbs().iterator();
+      while (breadcrumbsIterator.hasNext()) {
+        Breadcrumb breadcrumb = breadcrumbsIterator.next();
+        if ("react-native".equals(breadcrumb.getOrigin())) {
+          breadcrumbsIterator.remove();
+        }
+      }
+    }
```

Comment:
m: I'm not sure how testable `fetchNativeDeviceContexts` is at the moment, but can we add tests for this method. If it requires refactor first lets do it in a follow up PR.

Reply:
Good idea @krystofwoldrich 👍 
I did a small refactoring and added tests with https://github.com/getsentry/sentry-react-native/pull/4253